### PR TITLE
QFieldCloud integration minor fixes

### PR DESCRIPTION
**Pull request by suricactus**
Minor, but timeconsuming...

Check the commit messages to see what have changed.

### CODE REVIEW

File: qfieldsync/gui/cloud_projects_dialog.py [hunk 1]
```diff
@@ -804,7 +803,7 @@ def select_local_dir(self) -> Optional[str]:
         initial_path = (
             self.localDirLineEdit.text()
             or str(Path(QgsProject.instance().homePath()).parent)
-            or self.default_local_dir
+            or self.preferences.value("cloudDirectory")
```
Comment (nirvn) [inline qfieldsync/gui/cloud_projects_dialog.py:806]: Excellent.

File: qfieldsync/core/cloud_api.py [hunk 2]
```diff
@@ -610,20 +610,6 @@ def projects(self) -> Optional[List[CloudProject]]:
     def error_reason(self) -> str:
         return self._error_reason
 
-    @property
-    def is_currently_open_project_cloud_local(self) -> bool:
```
Comment (nirvn) [inline qfieldsync/core/cloud_api.py:614]: @suricactus , it is _not_ the same as currently_open_project. The latter *requires* a connected state and a valid project cache, whereas this function only checks whether the local directory is registered as a mirror of a server cloud project.

Without this, the qfieldsync toolbar's cloud synchronization button can't be enabled / disabled properly when not connected, which isn't great UX.

File: qfieldsync/qfield_sync.py [hunk 3]
```diff
@@ -145,6 +145,11 @@ def __init__(self, iface):
             self.cloud_item_gui_provider
         )
 
+        # autologin
+        dialog = CloudLoginDialog(self.network_manager)
```
Comment (nirvn) [inline qfieldsync/qfield_sync.py:150]: -1 to auto login. It's intrusive, and potentially a security risk in some environment. 

If we think an auto login is useful for some, let's lock that behavior behind a [x] auto login on QGIS launch setting.

File: qfieldsync/core/cloud_api.py (hunk 2 above)
Comment (suricactus) [reply qfieldsync/core/cloud_api.py:614]: Ah, I see now. You are right. A comment can help next time to highlight the difference.

Comment (suricactus) [reply qfieldsync/core/cloud_api.py:614]: What I see as a problem is there might be a `project_id` configuration in the local paths that no longer exists.

Comment (suricactus) [reply qfieldsync/core/cloud_api.py:614]: Added docs and a check for the project_id, if projects are present

File: qfieldsync/qfield_sync.py (hunk 3 above)
Comment (suricactus) [reply qfieldsync/qfield_sync.py]: I will open another PR with this.

Comment (suricactus) [reply qfieldsync/qfield_sync.py]: see https://github.com/opengisch/qfieldsync/pull/290

**Pull request merged**